crates/core/app/tests/mock_consensus.rs

```diff
@@ -80,54 +87,90 @@ async fn mock_consensus_can_send_a_spend_action() -> anyhow::Result<()> {
     let guard = common::set_tracing_subscriber();
     let storage = TempStorage::new().await?;
     let mut test_node = common::start_test_node(&storage).await?;
-    let mut rng = <rand_chacha::ChaChaRng as rand_core::SeedableRng>::seed_from_u64(0xBEEF);
 
     // Sync the mock client, using the test account's full viewing key, to the latest snapshot.
     let (viewing_key, spend_key) = (&test_keys::FULL_VIEWING_KEY, &test_keys::SPEND_KEY);
     let client = MockClient::new(viewing_key.deref().clone())
         .with_sync_to_storage(&storage)
-        .await?;
+        .await?
+        .tap(|c| info!(client.notes = %c.notes.len(), "mock client synced to test storage"));
 
     // Take one of the test account's notes...
     let (commitment, note) = client
         .notes
         .iter()
         .next()
         .ok_or_else(|| anyhow!("mock client had no note"))?
-        .tap(|(commitment, note)| {
-            tracing::info!(?commitment, ?note, "mock client note commitment")
-        });
+        .tap(|(commitment, note)| info!(?commitment, ?note, "mock client note commitment"));
 
-    // Build a transaction spending this note.
-    let tx = {
-        let position = client
+    let spend: ActionPlan = {
+        let proof = client
             .sct
-            .witness(*commitment)
-            .ok_or_else(|| anyhow!("commitment is not witnessed"))?
-            .position();
-        let spend = SpendPlan::new(&mut rng, note.clone(), position);
+            .witness(commitment.clone())
+            .ok_or_else(|| anyhow!("commitment is not witnessed"))?;
+        let note = note.clone();
+        let position = proof.position();
+        SpendPlan::new(&mut OsRng, note, position).into()
+    };
+
+    // Build a transaction spending this note.
+    let tx: Transaction = {
```

Comment:
Rather than doing this, I think it would be cleaner to write
```
let plan = TransactionPlan {
...
};
let tx = plan.build_concurrent().await;
```
The focus should be on the plan, the transaction is a side effect.
```

Reply:
tidied in 54c0b35d8 , it seemed easier to do it by example than to do line-by-line

Reply:
you resolved this in https://github.com/penumbra-zone/penumbra/pull/3891/commits/54c0b35d82802c08b3a48cc2a1008c69951eaa9b.